### PR TITLE
Adding the latest fixes and features to flutter_web_auth

### PR DESCRIPTION
**Pull request by Silfalion**
This pull request includes:

- On the web, calling on flutter_web_auth now opens a new tab instead of being an in-tab replacement.
- Updated android dependencies.
- Added working example for the web.

The current issue with the example is that it short-circuits the process.

Normally it would be:

- tap on button
- opens a separate HTML page with a verify button.
- tap verify button. 
- Takes you back to the original page with the wanted data.

This doesn't work on flutter web because this would need a separate server to host the HTML page and flutter web doesn't support that. So instead the tap button navigates directly to the URL that the verify button from the second page contains.

This opens a new Tab then closes it immediately. The information still arrives to the original page.

This isn't a problem when using the package directly, as I've tested before.

### CODE REVIEW

**Comment by christyjacob4** [comment]
@lohanidamodar  Please review this when you get the time

File: example/android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,36 +1,24 @@
...
-        android:icon="@mipmap/ic_launcher">
-        <activity
-            android:name=".MainActivity"
-            android:launchMode="singleTop"
-            android:theme="@style/LaunchTheme"
-            android:configChanges="orientation|keyboardHidden|keyboard|screenSize|locale|layoutDirection|fontScale|screenLayout|density|uiMode"
-            android:hardwareAccelerated="true"
-            android:windowSoftInputMode="adjustResize">
+    <application android:name="io.flutter.app.FlutterApplication" android:label="flutter_web_auth_example" android:icon="@mipmap/ic_launcher">
+        <activity android:name=".MainActivity" android:launchMode="singleTop" android:theme="@style/LaunchTheme" android:configChanges="orientation|keyboardHidden|keyboard|screenSize|locale|layoutDirection|fontScale|screenLayout|density|uiMode" android:exported="true" android:hardwareAccelerated="true" android:windowSoftInputMode="adjustResize">
```
Comment (lohanidamodar) [inline example/android/app/src/main/AndroidManifest.xml]: Can we fix code formatted with proper link breaks

File: example/pubspec.yaml [hunk 2]
```diff
@@ -11,7 +11,7 @@ dependencies:
 
   # The following adds the Cupertino Icons font to your application.
   # Use with the CupertinoIcons class for iOS style icons.
-  cupertino_icons: ^0.1.2
+  cupertino_icons: ^1.0.3
```
Comment (lohanidamodar) [inline example/pubspec.yaml]: Don't think this is used anywhere in the example project, can we remove this?

File: example/test/widget_test.dart [hunk 3]
```diff
@@ -0,0 +1,31 @@
+// This is a basic Flutter widget test.
```
Comment (lohanidamodar) [inline example/test/widget_test.dart]: should remove this as this is not a valid test for the example

**Review by lohanidamodar — CHANGES_REQUESTED** [review]
Great work @Silfalion , few changes. Will need to test and review once more!

File: example/pubspec.yaml (hunk 2 above)
Comment (Silfalion) [reply example/pubspec.yaml]: True. Resolved!

File: example/test/widget_test.dart (hunk 3 above)
Comment (Silfalion) [reply example/test/widget_test.dart]: Deleted!

File: example/android/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (Silfalion) [reply example/android/app/src/main/AndroidManifest.xml]: And fixed:)

**Comment by Silfalion** [comment]
Thank you! Once we're done here, I can push a PR for the flutter SDK generator as well, though it will need more thorough reviewing. It kinda worked when I tested so it shouldn't be too far off the mark.

**Comment by lohanidamodar** [comment]
> Thank you! Once we're done here, I can push a PR for the flutter SDK generator as well, though it will need more thorough reviewing. It kinda worked when I tested so it shouldn't be too far off the mark.

And before the PR in SDK generator can be finalized we have to publish this as a new package with a good name along with proper attribution back to the original package.

File: README.md [hunk 4]
```diff
@@ -70,7 +70,7 @@ final accessToken = jsonDecode(response.body)['access_token'] as String;
 
 ## Setup
 
-Setup works as for any Flutter plugin, expect the Android caveat outlined below:
+Setup works as for any Flutter plugin, expect the Android and Web caveats outlined below:
```
Comment (lohanidamodar) [inline README.md]:
```suggestion
Setup works as it is for any Flutter plugin, expect the Android and Web caveats outlined below:
```

File: example/android/app/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -1,36 +1,26 @@
...
-        android:label="flutter_web_auth_example"
-        android:icon="@mipmap/ic_launcher">
-        <activity
-            android:name=".MainActivity"
-            android:launchMode="singleTop"
-            android:theme="@style/LaunchTheme"
-            android:configChanges="orientation|keyboardHidden|keyboard|screenSize|locale|layoutDirection|fontScale|screenLayout|density|uiMode"
-            android:hardwareAccelerated="true"
-            android:windowSoftInputMode="adjustResize">
+    <application android:name="io.flutter.app.FlutterApplication" android:label="flutter_web_auth_example" android:icon="@mipmap/ic_launcher">
```
Comment (lohanidamodar) [inline example/android/app/src/main/AndroidManifest.xml]:
```suggestion
    <application
    	android:name="io.flutter.app.FlutterApplication"
    	android:label="flutter_web_auth_example"
    	android:icon="@mipmap/ic_launcher">
```
I meant something like this for all to make it readable like before

File: example/lib/home_page.dart [hunk 6]
```diff
@@ -0,0 +1,145 @@
+import 'dart:io';
```
Comment (lohanidamodar) [inline example/lib/home_page.dart]:
```suggestion
import 'dart:io' show HTTPServer;
```
Let's only use what we need.

**Review by lohanidamodar — CHANGES_REQUESTED** [review]

File: example/lib/home_page.dart (hunk 6 above)
Comment (Silfalion) [reply example/lib/home_page.dart]: Pretty instructive 🤔

File: example/android/app/src/main/AndroidManifest.xml (hunk 5 above)
Comment (Silfalion) [reply example/android/app/src/main/AndroidManifest.xml]: Ah I see. It’s better indeed.

**Comment by Silfalion** [comment]
Lol guys, LinusU merged the repo. So there is no need for us to make a repo anymore.

**Pull request closed without merging**